### ISSUE TITLE
support custom resource extensions in `<script>` and `<style>` tags

### ISSUE THREAD

**Issue by thescientist13** [ProjectEvergreen/greenwood#924]
## Type of Change
- [x] New Feature Request
- [ ] Documentation / Website
- [ ] Improvement / Suggestion
- [ ] Bug
- [ ] Other (please clarify below)

## Summary
Observed in https://github.com/ProjectEvergreen/greenwood-getting-started/pull/54 that although resource plugins allow using custom extensions for ESM, like for TypeScript or SCSS in JavaScript using `import`, but this behavior is not supported in `<script>` and `<style>` tags.

## Details
So basically, you can do this
```js
import { FooterComponent } from './footer.ts';

// or this

import css from './styles.scss?type=css';
```

But we need to make sure user's can also do this
```html
<script type="module" src="./footer.ts"/>


<link rel="stylesheet" href="./styles.scss"/>
```

**Comment by thescientist13** [comment]
I think this might already be working?  Seems like it works just fine for #1085 , maybe it was already resolved as part of #971 or something otherwise released as part of [**v0.27.0**](https://github.com/ProjectEvergreen/greenwood/releases/tag/v0.27.0)?  🤔

**Comment by thescientist13** [comment]
yeah, seems fine?  Another example here
https://github.com/AnalogStudiosRI/www.analogstudios.net/blob/main/src/index.html#L26

Going to close this and backtag this to v0.27.0 for now.
